eva/udfs/decorators/io_descriptors/abstract_types.py

```diff
@@ -41,6 +42,47 @@ def generate_catalog_entries(
         """
         pass
 
+    @abstractmethod
+    def check_array_and_convert_shape(self, input_obj):
+        pass
+
+    @abstractmethod
+    def check_array_and_convert_type(self, input_obj, data_type):
+        pass
+
+    def is_array_type_defined(self):
+        if self.array_type is None:
+            return False
+        return True
+
+    def is_shape_defined(self):
+        if self.array_dimensions is None:
+            return False
+        return True
+
+    def validate_object(self, object, input_flag=True):
+        if self.is_array_type_defined:
+            try:
+                object = self.check_array_and_convert_type(object)
+            except Exception as e:
+                if input_flag:
+                    msg = "Data type mismatch of Input parameter. "
+                else:
+                    msg = "Data type mismatch of Output parameter."
+                raise UDFIODefinitionError(msg)
```

Comment:
Moving to assert syntax will mostly cleanup code by avoiding try and except everywhere.

Reply:
The assert statement can have the appropriate message.

Reply:
same response as before, the shape conversions might throw an error if its not possible, hence used try-except